src/redash_stmo/data_sources/health.py

```diff
@@ -88,16 +95,25 @@ def test_connection(self, custom_query_text=None):
 @require_super_admin
 def stmo_status_api():
     status = original_get_status()
-    status['data_sources'] = json.loads(redis_connection.get('data_sources:health') or '{}')
+    health_data = json.loads(redis_connection.get('data_sources:health') or '{}')
+
+    # Get the top level status for each data source
+    for data_source_id in health_data:
+        data_source_name = health_data[data_source_id]["metadata"]["name"]
+        dashboard_label = "[Data Source Health] {name}".format(name=data_source_name)
+        status[dashboard_label] = health_data[data_source_id]["status"]
     return jsonify(status)
 
 
-def datasource_health(app=None):
+def health(app=None):
     """A Redash extension to add datasource health status reporting."""
 
     # Override the default status API view with our extended view
     app.view_functions['%s.status_api' % routes.name] = stmo_status_api
 
+    # Add a new endpoint with full health data
+    add_resource(app, DataSourceHealthResource, '/data_source_health.json')
```

Comment:
Let's use a namespace for this URL endpoint, assuming we get more of those:
```suggestion
    add_resource(app, DataSourceHealthResource, '/status/data_sources/health.json')
```